### ISSUE TITLE
Cannot stop build events from being run via dotnet build or using conditions in .csproj

### ISSUE THREAD

**Issue by mojoex** [dotnet/project-system#4455]
We have Post-Build events that execute a CMD file and then uses gulp to compile some typescript files.
We have a requirement that the Post-Build events should only trigger when the solution or Project is built inside VS2017.

As I understand it, [https://github.com/dotnet/project-system/pull/2367](https://github.com/dotnet/project-system/pull/2367) outlined a change from a Property to a Target, which seems to have disabled this functionality.

Without the property, adding "-p:PostBuildEvent=" or "/p:PostBuildEvent=" at the end of a dotnet build command does not provide this behaviour.

Using a condition in the Target style entry also does not seem to trigger at any point.

At the moment - we see no way of being able to exclude PostBuildEvents when building a project that has them defined. Is this correct? If not, any guidance you could provide would be great.

**Comment by davidwengier** [comment]
> Using a condition in the Target style entry also does not seem to trigger at any point.

Can you include a sample of the project file you are using? The following seems to work for me:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

  <Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition="'$(BuildingInsideVisualStudio)' == 'true'">
    <Exec Command="echo Hi There!" />
  </Target>

</Project>
```

It correctly outputs "Hi there!" in VS2017, but not when using msbuild from the command line:
![image](https://user-images.githubusercontent.com/754264/51008764-4438c580-15a1-11e9-8cb7-04f7cee0e8ff.png)

**Comment by mojoex** [comment]
Thanks for the reply David!
I actually ended up approaching this in a different way entirely and had two options in the end.

So we originally had this;

`  <Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition="'$(BuildingInsideVisualStudio)' == '' Or '$(BuildingInsideVisualStudio)' == 'true'">
<Exec Command="do stuff" />
</Target>`

And we couldn't get a "dotnet build" CLI to ignore the PostBuildEvents with this.

So the 1st of the other options was to just go back to the PropertyGroup method of applying a PostBuildEvent. This allowed the properties to be set in dotnet CLI with "-p:PostBuildEvent=" and fit the purpose.

But after reviewing the solution we realised that a better option would be to try and only run the events on a Rebuild in VS. So we used targets for "OnCleanOnly" which is run during a Rebuild.

So we ended up with;

`  <Target Name="OnCleanOnly" AfterTargets="Clean">
    <Exec Command="do stuff" />
  </Target>`

This fits us much better as the PostBuildEvent is a GitHook and Gulp compile for dependant files that aren't included in the solution. So we only needed to ensure that we build our TS files when we pull down a new commit or switch branch.
The dotnet build CLI command is used in a testing solution and added a lot of time when we build the dependant projects before running the tests. So the OnCleanOnly target suits us just fine.
The main issue we had was that setting the properties in the dotnet CLI is not supported when using the Target method of build events.

**Comment by davidwengier** [comment]
> setting the properties in the dotnet CLI is not supported when using the Target method of build events

Unfortunately, moving from properties to targets was a deliberate, and required, change for the new project system.

Glad you found a solution that works for you 😁

**Comment by nimblejoe** [comment]
@mojoex thank you for the tip.  I followed your concept above and moved Post-Build event to Rebuild.  However, instead of "Clean", I had to use "Rebuild" <Target Name="OnRebuild" AfterTargets="Rebuild"> <Exec Command="do stuff" /> </Target>.  The problem I had to solve is that locally I'm building in VS on a Windows workstation but in my pipeline I'm doing a Docker build on a Linux machine which causes challenges and couldn't get beyond.  This worked.
